src/main.rs

```diff
@@ -47,125 +87,245 @@ impl LogLevel {
 }
 
 #[derive(Subcommand)]
+/// params that greenboot accepts
+///
+/// greenboot health-check -> runs the custom health checks
+///
+/// greenboot rollback -> if boot_counter satisfies it trigger rollback
 enum Commands {
-    Check,
-    Stamp,
-}
-
-#[derive(Serialize, Deserialize, PartialEq, Eq, Hash)]
-struct ServiceStatus {
-    unit: String,
+    HealthCheck,
+    Rollback,
 }
 
-fn check() -> Result<(), Error> {
-    match File::open("/etc/greenboot/upgrade.stamp") {
-        Ok(_) => {
-            log::info!("stamp on disk, removing and running greenboot");
-            std::fs::remove_file("/etc/greenboot/upgrade.stamp")?
+/// this runs the scripts in required.d and wanted.d
+fn run_diagnostics() -> Result<(), Error> {
+    let mut script_failure: bool = false;
+    let mut path_exists: bool = false;
+    for path in GREENBOOT_INSTALL_PATHS {
+        let greenboot_required_path = format!("{path}/check/required.d/");
+        if !Path::new(&greenboot_required_path).is_dir() {
+            continue;
         }
-        Err(e) => match e.kind() {
-            ErrorKind::NotFound => return Ok(()),
-            _ => {
-                bail!("unknown error when opening stamp file: {:?}", e);
-            }
-        },
-    }
-    let mut failure = false;
-    for path in [
-        "/usr/lib/greenboot/check/required.d/*.sh",
-        "/etc/greenboot/check/required.d/*.sh",
-    ] {
-        for entry in glob(path)?.flatten() {
+        path_exists = true;
+        let greenboot_required_path = format!("{greenboot_required_path}*.sh");
+        for entry in glob(&greenboot_required_path)?.flatten() {
             log::info!("running required check {}", entry.to_string_lossy());
-            let output = Command::new("bash").arg("-C").arg(entry).output()?;
+            let output = Command::new("bash").arg("-C").arg(entry.clone()).output()?;
             if !output.status.success() {
-                // combine and print stderr/stdout
-                log::warn!("required script failed...");
-                failure = true;
+                log::error!("required script {} failed!", entry.to_string_lossy());
+                log::error!("reason: {}", String::from_utf8_lossy(&output.stderr));
+                script_failure = true;
             }
         }
     }
-    for path in [
-        "/usr/lib/greenboot/check/wanted.d/*.sh",
-        "/etc/greenboot/check/wanted.d/*.sh",
-    ] {
-        for entry in glob(path)?.flatten() {
-            log::info!("running required check {}", entry.to_string_lossy());
-            let output = Command::new("bash").arg("-C").arg(entry).output()?;
+
+    if !path_exists {
+        bail!("required.d not found");
```

Comment:
I believe this error can be enhanced - something like 'the required.d folder doesn't exist'

Reply:
>  believe the code is flacky, what happens if in the future we add a third path w/o changing the code here?
by updating the variable
```
/// dir that greenboot looks for the health check and other scripts
static GREENBOOT_INSTALL_PATHS: [&str; 2] = ["/usr/lib/greenboot", "/etc/greenboot"];
```

> I believe this error can be enhanced - something like 'the required.d folder doesn't exist'

I think "cannot find any required.d folder"  makes more sense as it will trigger if only when none of the required.d folders does not exist

Reply:
fixed in : ee8bd08b11f22f930ca7e0e7a43733cbd28dd8b2